setup.sh

```diff
@@ -61,11 +63,21 @@ function _preparation() {
   fi
   apt-get -q -y update >> ${log} 2>&1
   apt-get -q -y upgrade >> ${log} 2>&1
-  apt-get -q -y install whiptail git sudo curl wget lsof fail2ban apache2-utils vnstat tcl tcl-dev build-essential dirmngr apt-transport-https python-pip >> ${log} 2>&1
+
+  if [[ $codename =~ ("focal") ]]; then 
+      pipv='python3-pip'
+      else
+      pipv='python-pip'
+  fi
+
+  echo "Installing dependencies"
+  # this apt-get should be checked and handled if fails, otherwise the install borks. 
```

Comment:
This command *should* be bulletproof, but there are times when this command will fail due to apt locks as I mentioned above and it would be best to avoid this scenario as the installer will continue along blissfully unaware that needed packages are missing.